source/load-balancers/load-balancers.rst

```diff
@@ -0,0 +1,535 @@
+response (i.e. the cursor is drained). When the cursor's :code:`close` method is
+invoked, either explicitly or via an implicit resource cleanup mechanism, the driver
+MUST use the same connection to execute a :code:`killCursors` command if necessary
+and then check the connection back into the pool regardless of the result.
+
+If a :code:`getMore` fails with a network error, drivers MUST leave the connection pinned
+to the cursor. When the cursor's :code:`close` method is invoked, drivers MUST NOT execute
+a :code:`killCursors` command because the pinned connection is no longer valid and MUST
+return the connection back to the pool.
+
+Behaviour With Transactions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When executing a transaction in load balancing mode, drivers MUST follow the rules outlined
+in `Sharded Transactions <../transactions/transactions.rst#sharded-transactions>`__ with one
+exception: drivers MUST use the same connection for all commands in the transaction
+(excluding retries of commitTranscation and abortTransaction in some cases). Pinning
+to a single connection ensures that all commands in the transaction target the same
+service behind the load balancer. The rules for pinning to a connection and releasing
+a pinned connection are the same as those for server pinning in non-load balanced sharded
+transactions as described in `When to unpin <../transactions/transactions.rst#when-to-unpin>`__.
+
+Connection Tracking
+^^^^^^^^^^^^^^^^^^^
+
+The driver connection pool MUST track the purpose for which connections are checked out
+in the following 3 categories:
+
+- Connections checked out for cursors
+- Connections checked out for transactions
+- Connections checked out for operations not falling under the previous 2 categories
+
+When the connection pool's :code:`maxPoolSize` is reached and the pool times out waiting
+for a new connection the :code:`WaitQueueTimeoutError` MUST include a new detailed message,
+"Timeout waiting for connection from the connection pool. maxPoolSize: n, connections
+in use by cursors: n, connections in use by transactions: n, connections in use by other
+operations: n". A :code:`ConnectionCheckOutFailedEvent` MUST also be published with the
+same message.
+
+
+Error Handling
+--------------
+
+Initial Handshake Errors
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+When establishing a new connection in load balanced mode, drivers MUST NOT perform SDAM
+error handling for any errors that occur before the MongoDB Handshake
+(i.e. :code:`hello` or legacy hello command) is complete. Errors during the MongoDB
+Handshake MUST also be ignored for SDAM error handling purposes. Once the initial
+handshake is complete, the connection MUST determine its generation number based
+on the :code:`serviceId` field in the handshake response. Any errors that occur
+during the rest of connection establishment (e.g. errors during authentication commands)
+MUST go through the SDAM error handling flow but MUST NOT mark the server as
```

Comment:
Forgot to include this in my previous suggestion, but we should also say that only connections with matching `serviceId` should be cleared if a pool clearing is required as part of the SDAM error handling process. The `Post-Handshake Errors` section correctly mentions this so you can match that.

Reply:
Included this in this paragraph now.